ql/src/test/queries/clientpositive/partition_coltype_literals.q

```diff
@@ -73,11 +73,11 @@ drop table if exists partcoltypeothers;
 create table partcoltypeothers (key int, value string) partitioned by (decpart decimal(6,2), datepart date);
 
 set hive.typecheck.on.insert=false;
-insert into partcoltypeothers partition (decpart = 1000.01BD, datepart = date '2015-4-13') select key, value from src limit 10;
+insert into partcoltypeothers partition (decpart = 1000.01BD, datepart = date '2015-04-13') select key, value from src limit 10;
```

Comment:
When disable `hive.typecheck.on.insert`, we need ensure the partition value is in right format.

Reply:
I don't agree with this change. This would be a breaking change if users were to use this patch in their production env (because they would now have to change their scripts/data). We intentionally specified this config 'hive.typecheck.on.insert=false' so that regardless of data format, partitions are created the way the data is inserted.

Reply:
The `hive.typecheck.on.insert` config is set to `true` by default, meaning that this change won't impact the most tasks. If users utilizes the date`2015-5-15` and also set `hive.typecheck.on.insert` to false, they can conveniently enable this config to resolve the issue. This behavior aligns with the config description.So I think this change is OK.